chain/db/db.go

```diff
@@ -10,18 +12,74 @@ import (
 
 type BandDB struct {
 	db *gorm.DB
+	tx *gorm.DB
 }
 
 func NewDB(dialect, path string) (*BandDB, error) {
 	db, err := gorm.Open(dialect, path)
-	db.CreateTable(Event{})
-
 	if err != nil {
 		return nil, err
 	}
+
+	db.AutoMigrate(&Metadata{}, &Event{})
+
 	return &BandDB{db: db}, nil
 }
 
+func (b *BandDB) SaveChainID(chainID string) {
+	var chainIDRow Metadata
+	// chainIDRow := Metadata{Key: "chain-id", Value: chainID}
+	b.tx.Where(Metadata{Key: "chain-id"}).Assign(Metadata{Value: chainID}).FirstOrCreate(&chainIDRow)
+}
+
+func (b *BandDB) ValidateChainID(chainID string) error {
+	var chainIDRow Metadata
+	b.tx.Where("key = ?", "chain-id").First(&chainIDRow)
+	if chainIDRow.Value != chainID {
+		return errors.New("Chain id not match")
+	}
+	return nil
+}
+
+func (b *BandDB) OpenTransaction() {
+	if b.tx != nil {
+		panic("There is an transaction that left open")
```

Comment:
`panic("BeginTransaction: Cannot begin a new transaction without closing the pending one.")`